tests/callbacks_test.cpp

```diff
@@ -8,6 +8,8 @@
  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
  */
 
+#include <limits>
```

Comment:
Hm, `ensmallen.hpp` includes `climits`, what about we remove `limits` here and put that into `ensmallen.hpp` instead and remove the `climits` header.

Reply:
Agreed :+1: